### PR TITLE
TandemFoil: T_max=10 + 4L/192d — deeper model at original width

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The TandemFoil baseline uses a 3L/192d model (3 layers, 192 hidden dim, 3 heads) with T_max=10 cosine annealing. On AirfRANS, moving from 3L/192d to 4L/256d gave significant gains. We hypothesize that adding a fourth layer at the same width (4L/192d) will improve TandemFoil performance without increasing the parameter count as aggressively as going to 256d. A deeper model can learn more complex inter-foil pressure interactions at the cost of only ~33% more compute per forward pass. This also keeps the hidden dim at 192 (3 heads of 64), which the optimizer is already well-tuned for.

4L/192d has:
- 4 transformer layers (vs 3 in baseline)
- 192 hidden dim, 3 attention heads (unchanged)
- ~33% more parameters in the transformer body

## Instructions
Run the TandemFoil training with a 4-layer, 192-dim model and T_max=10 cosine annealing. Keep all other hyperparameters identical to the baseline:

```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 3e-4 \
  --cosine_t_max 10 \
  --num_layers 4 \
  --hidden_dim 192 \
  --num_heads 3 \
  --slices 64 \
  --epochs 999 \
  --wandb_group tandem-4L192d-tmax10
```

If `--num_layers`, `--hidden_dim`, or `--num_heads` flags don't exist, check train.py for the correct flag names and adjust accordingly. The goal is a 4-layer transformer at 192 hidden dim with 3 heads.

Report val_primary/surface_pressure_mae and compare against baseline 75.59.

## Baseline
Current best metrics from BASELINE.md:
- val_primary/surface_pressure_mae: **75.59**
- W&B run: 77yoba65
- PR: #2490 (TandemFoil: T_max=10, Lion lr=3e-4, 3L/192d, slices=64, 14 epochs)
- Reproduce command: `cd target/icml2026 && python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine_t_max 10 --num_layers 3 --hidden_dim 192 --num_heads 3 --slices 64 --epochs 999`

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing — 4L/192d is obsolete. 5L/256d achieves 52.81 (PR #2595). kakashi, please pick up your next assignment.
